### PR TITLE
Split task queues into jobs

### PR DESCRIPTION
**Pull request by sunu**

### CODE REVIEW

File: servicelayer/util.py [hunk 1]
```diff
@@ -15,3 +16,19 @@ def backoff(failures=0):
 def service_retries():
     """A default number of tries to re-try an external service."""
     return range(30)
```
Comment (pudo) [inline servicelayer/util.py]: Should we move the other helper functions from `memorious.model.common` here so they are all in one place?

File: servicelayer/process.py [hunk 2]
```diff
@@ -22,13 +23,14 @@ class ServiceQueue(object):
     PRIO_LOW = 1
     PRIORITIES = [PRIO_HIGH, PRIO_MEDIUM, PRIO_LOW]
 
-    def __init__(self, conn, operation, dataset, priority=PRIO_MEDIUM):
+    def __init__(self, conn, operation, job_id, dataset, priority=PRIO_MEDIUM):
         self.conn = conn
         self.operation = operation
+        self.job_id = job_id
```
Comment (pudo) [inline servicelayer/process.py]: Should we have a helper function for making a good job id (basically just uuid4?). Could be `Job.random_id()`.

File: servicelayer/process.py [hunk 3]
```diff
@@ -47,14 +50,30 @@ def task_done(self):
         self.progress.mark_finished()
```
Comment (pudo) [inline servicelayer/process.py]: I wonder if we should add an automatic test here to say: if done, remove the job/op?

**Comment by pudo** [comment]
Do you feel like a `job_instance.get_tasks(limit=500)` function could still fit in here? You'd basically have to get many items from the list and remove them in the same moment via an atomic pipeline.

**Comment by sunu** [comment]
> Do you feel like a `job_instance.get_tasks(limit=500)` function could still fit in here? You'd basically have to get many items from the list and remove them in the same moment via an atomic pipeline.

I have added this method. But what's the use case for this?

**Pull request merged**